src/spikeinterface/preprocessing/unsigned_to_signed.py

```diff
@@ -46,7 +48,11 @@ def get_traces(self, start_frame, end_frame, channel_indices):
         traces_dtype = traces.dtype
         nbits = traces_dtype.itemsize * 8
         signed_dtype = f"int{2 * (traces_dtype.itemsize) * 8}"
-        offset = 2 ** (nbits - 1)
+        if self.bit_depth is not None:
+            offset = 2 ** (self.bit_depth - 1)
+        else:
+            # offset = 2 ** (nbits - 1)
+            offset = 0
```

Comment:
```suggestion
            offset = 2 ** (nbits - 1)
```
The old behavior should be default. In general, we can assume ADCs are used full range